File: backend/app/Models/SalesInvoiceItem.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class SalesInvoiceItem extends Model
{
    protected $fillable = ['sales_invoice_id', 'batch_id', 'qty', 'price'];

    public function invoice()
    {
        return $this->belongsTo(SalesInvoice::class, 'sales_invoice_id');
    }

    public function batch()
    {
        return $this->belongsTo(Batch::class);
    }
}

File: desktop/lib/features/sales/domain/entities/sales_invoice_item.dart
class SalesInvoiceItem {
  final String id;
  final String salesInvoiceId;
  final String batchId;
  final String? batchNumber;
  final String? medicineName;
  final int quantity;
  final int price;

  const SalesInvoiceItem({
    required this.id,
    required this.salesInvoiceId,
    required this.batchId,
    this.batchNumber,
    this.medicineName,
    required this.quantity,
    required this.price,
  });

  double get total => (quantity * price).toDouble();
}
